### DISCUSSION TITLE
how to make the stimulus disappear on keyboard hit

### DISCUSSION THREAD

**Discussion by momolinus** [jspsych/jsPsych#1661] [Q&A] [answered]
Hi,

I have a problem with image-keyboard-response:

```javascript
let trial = {
      type: 'image-keyboard-response',
      stimulus: 'stimulus.jpg',
      choices: ["2"],
      prompt: "<p>[Press 2]</p>",
      response_ends_trial: false,
      trial_duration: 5000,
      stimulus_duration: null
};
```
I thought that with this configuration I would achieve that the stimulus disappears when a user makes the correct keystroke. Instead of the stimulus, an empty area should be displayed. Unfortunately, the stimulus is displayed for the entire duration of the trial.

Is there a configuration where the stimulus disappears on keystroke but the trial does not end yet.

momolinus

**Comment by trig11** [comment, marked as answer]
Hi, 
I think the documentation might be a bit ambigous at that point. To achieve what you want to do, I think you will need to define the CSS-class "responded". As you will see in the plugin code, this is going to change the image display after a response has been given.

If you want to remove the image after a response, you can do it via CSS:
```
.responded{
	visibility: hidden;
}
```

If you have one, this will likely leave the prompt on screen, though. In order to also change the prompt, you will first need to give your prompt a class (e.g., the class "prompttext" and then modify the after_response behavior of the plugin a bit. You can add this in there:
```
document.getElementsByClassName("prompttext")[0].innerHTML  = '<p> New prompt text here </p>'; 
```
